src/Data/Singletons/Internal.hs

```diff
@@ -56,8 +56,15 @@ class SingI (a :: k) where
   -- extension to use this method the way you want.
   sing :: Sing a
 
--- | Permits pattern matching on an explicit @Sing a@ value to bring an
--- implicit @SingI a@ constraint into scope.
+-- | A pattern synonym for implicit singletons.
+--
+-- As a pattern: We can match on an explicit @Sing a@ witness to bring
+-- an implicit @SingI a@ constraint into scope.
+--
+-- As a constructor: It constructs a singleton @Sing a@ as long as we
+-- have an implicit singleton constraint @SingI a@.
+--
+-- /Since: 2.4/
```

Comment:
There's not much point in putting `Since` annotations in this library, since we completely break backwards compatibility every release.

Reply:
It is often useful to me to know when a feature was added, I'll remove it for consistency